crates/c-api/src/table.rs

```diff
@@ -67,7 +67,7 @@ pub unsafe extern "C" fn wasm_table_get(
     index: wasm_table_size_t,
 ) -> Option<Box<wasm_ref_t>> {
     let table = t.table();
-    let r = table.get(t.ext.store.context_mut(), index)?;
+    let r = table.get(t.ext.store.context_mut(), index as u64)?;
```

Comment:
Where possible it's generally recommended to use `t::from(val)` instead of `val as t` since the former is guaranteed to be lossless where the latter can accidentally lose bits. Can this be changed to `u64::from(index)`? (here and some places below too)